core/scripts.py

```diff
@@ -2861,16 +2846,28 @@ def process_task(self, task: DeferredTask):
             for file in files:
                 os.remove(file.name)
 
-    def generate_report(self, data_source_name: str, library_id: int) -> str:
-        """Generate a csv file and return the file path"""
-        with tempfile.NamedTemporaryFile(
-            "w",
-            delete=False,
-        ) as temp:
+    def prepare_email_attachment(self, attachments, file_path, attachment_name, files):
```

Comment:
Minor: Missing type hints on this function

Reply:
I'll fix.